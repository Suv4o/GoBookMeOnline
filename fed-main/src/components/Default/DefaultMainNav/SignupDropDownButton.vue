<script lang="ts">
export default {
  name: 'SignupDropDown',
}
</script>

<script setup lang="ts">
import { Popover, PopoverButton, PopoverPanel } from '@headlessui/vue'
// import useState from './useState'
import { ref } from 'vue'
import { Roles } from '../../../types/enums'

const button = ref<HTMLButtonElement>()

interface SignupOptions {
  name: string
  description: string
  href: string
  roleType: Roles
}

interface Props {
  signupOptions: SignupOptions[]
}

const props = withDefaults(defineProps<Props>(), {})

// const { setRoleType } = useState()

// function handleSignupOptionClick(roleType: Roles) {
//   setRoleType(roleType)
//   button.value?.click()
// }
</script>

<template>
  <Popover class="relative">
    <PopoverButton as="div">
      <button
        ref="button"
        class="inline-flex items-center px-4 py-2 ml-4 border border-transparent text-base font-medium rounded-md text-white bg-teal-600 hover:bg-teal-700 shadow"
      >
        Sign up
      </button>
    </PopoverButton>

    <transition
      enter-active-class="transition ease-out duration-200"
      enter-from-class="opacity-0 translate-y-1"
      enter-to-class="opacity-100 translate-y-0"
      leave-active-class="transition ease-in duration-150"
      leave-from-class="opacity-100 translate-y-0"
      leave-to-class="opacity-0 translate-y-1"
    >
      <PopoverPanel class="absolute left-[-48px] z-10 mt-3 w-screen max-w-xs -translate-x-1/2 transform px-2 sm:px-0">
        <div class="overflow-hidden rounded-lg shadow-lg ring-1 ring-black ring-opacity-5">
          <div class="relative grid gap-6 bg-white px-5 py-6 sm:gap-8 sm:p-8">
            <router-link
              v-for="option in props.signupOptions"
              :key="option.name"
              :to="{ name: option.href }"
              class="-m-3 block rounded-md p-3 transition duration-150 ease-in-out hover:bg-gray-50"
              @click="button?.click()"
            >
              <p class="text-base font-medium text-gray-900">{{ option.name }}</p>
              <p class="mt-1 text-sm text-gray-500">{{ option.description }}</p>
            </router-link>
          </div>
        </div>
      </PopoverPanel>
    </transition>
  </Popover>
</template>
