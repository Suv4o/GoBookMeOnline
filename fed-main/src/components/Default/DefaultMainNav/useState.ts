import { reactive, toRefs } from 'vue'

const state = reactive({
  isSignInButtonShown: true,
  isSignUpButtonShown: true,
})

export default () => {
  function setSignInButtonShown(isShown = true) {
    state.isSignInButtonShown = isShown
  }

  function setSignUpButtonShown(isShown = true) {
    state.isSignUpButtonShown = isShown
  }

  return {
    ...toRefs(state),
    setSignInButtonShown,
    setSignUpButtonShown,
  }
}
