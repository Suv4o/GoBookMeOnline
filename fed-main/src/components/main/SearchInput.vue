<script lang="ts">
export default {
  name: 'SearchInput',
}
</script>

<script setup lang="ts">
import { computed, ref } from 'vue'
import { SearchIcon } from '@heroicons/vue/solid'
import { ChevronRightIcon, UsersIcon } from '@heroicons/vue/outline'
import { Combobox, ComboboxInput, ComboboxOptions, ComboboxOption } from '@headlessui/vue'

const people = [
  {
    id: 1,
    name: 'Leslie Alexander',
    phone: '1-493-747-9031',
    email: 'lesliealexander@example.com',
    role: 'Co-Founder / CEO',
    url: 'https://example.com',
    profileUrl: '#',
    imageUrl:
      'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80',
  },
  {
    id: 2,
    name: 'Juniper Moody',
    phone: '1-582-300-8610',
    email: 'lesliealexander@example.com',
    role: 'Co-Founder / CEO',
    url: 'https://example.com',
    profileUrl: '#',
    imageUrl:
      'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80',
  },
  {
    id: 3,
    name: 'Miyah Brady',
    phone: '1-316-938-5949',
    email: 'lesliealexander@example.com',
    role: 'Co-Founder / CEO',
    url: 'https://example.com',
    profileUrl: '#',
    imageUrl:
      'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80',
  },
  {
    id: 4,
    name: 'Ellen Walker',
    phone: '1-203-368-8795',
    email: 'lesliealexander@example.com',
    role: 'Co-Founder / CEO',
    url: 'https://example.com',
    profileUrl: '#',
    imageUrl:
      'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80',
  },
  {
    id: 5,
    name: 'Emma Nicholson',
    phone: '1-248-393-7011',
    email: 'lesliealexander@example.com',
    role: 'Co-Founder / CEO',
    url: 'https://example.com',
    profileUrl: '#',
    imageUrl:
      'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80',
  },
]

// const recent = [people[0], people[1], people[2], people[3], people[4]]
const recent: any = []
const isFocused = ref(false)

const query = ref('')
const filteredPeople = computed(() =>
  query.value === ''
    ? []
    : people.filter(person => {
        return person.name.toLowerCase().includes(query.value.toLowerCase())
      })
)

function onSelect(person: any) {
  window.location = person.url
}

function setRecentSearch() {
  isFocused.value = true
}

function unsetRecentSearch() {
  isFocused.value = false
}
</script>

<template>
  <Combobox
    v-slot="{ activeOption }"
    as="div"
    class="transform divide-y divide-gray-100 overflow-hidden rounded-xl bg-white shadow-2xl ring-1 ring-black ring-opacity-5 transition-all"
    @update:model-value="onSelect"
  >
    <div class="relative">
      <SearchIcon class="pointer-events-none absolute top-3.5 left-4 h-5 w-5 text-gray-400" aria-hidden="true" />
      <ComboboxInput
        class="h-12 w-full border-0 bg-transparent pl-11 pr-4 text-gray-800 placeholder-gray-400 focus:ring-0 sm:text-sm"
        placeholder="Search..."
        @focus="setRecentSearch"
        @blur="unsetRecentSearch"
        @change="query = $event.target.value"
      />
    </div>

    <ComboboxOptions
      v-if="query === '' || filteredPeople.length > 0"
      class="flex divide-x divide-gray-100"
      as="div"
      static
      hold
    >
      <div
        v-if="isFocused && (query !== '' || recent.length)"
        :class="['max-h-96 min-w-0 flex-auto scroll-py-4 overflow-y-auto px-6 py-4', activeOption && 'sm:h-96']"
      >
        <h2 v-if="query === '' && recent.length" class="mt-2 mb-4 text-xs font-semibold text-gray-500">
          Recent searches
        </h2>
        <div hold class="-mx-2 text-sm text-gray-700">
          <ComboboxOption
            v-for="person in query === '' ? recent : filteredPeople"
            :key="person.id"
            v-slot="{ active }"
            :value="person"
            as="template"
          >
            <div
              :class="[
                'group flex cursor-default select-none items-center rounded-md p-2',
                active && 'bg-gray-100 text-gray-900',
              ]"
            >
              <img :src="person.imageUrl" alt="" class="h-6 w-6 flex-none rounded-full" />
              <span class="ml-3 flex-auto truncate">{{ person.name }}</span>
              <ChevronRightIcon v-if="active" class="ml-3 h-5 w-5 flex-none text-gray-400" aria-hidden="true" />
            </div>
          </ComboboxOption>
        </div>
      </div>

      <div
        v-if="activeOption"
        class="hidden h-96 w-1/2 flex-none flex-col divide-y divide-gray-100 overflow-y-auto sm:flex"
      >
        <div class="flex-none p-6 text-center">
          <img :src="activeOption.imageUrl" alt="" class="mx-auto h-16 w-16 rounded-full" />
          <h2 class="mt-3 font-semibold text-gray-900">
            {{ activeOption.name }}
          </h2>
          <p class="text-sm leading-6 text-gray-500">{{ activeOption.role }}</p>
        </div>
        <div class="flex flex-auto flex-col justify-between p-6">
          <dl class="grid grid-cols-1 gap-x-6 gap-y-3 text-sm text-gray-700">
            <dt class="col-end-1 font-semibold text-gray-900">Phone</dt>
            <dd>{{ activeOption.phone }}</dd>
            <dt class="col-end-1 font-semibold text-gray-900">URL</dt>
            <dd class="truncate">
              <a :href="activeOption.url" class="text-teal-600 underline">
                {{ activeOption.url }}
              </a>
            </dd>
            <dt class="col-end-1 font-semibold text-gray-900">Email</dt>
            <dd class="truncate">
              <a :href="`mailto:${activeOption.email}`" class="text-teal-600 underline">
                {{ activeOption.email }}
              </a>
            </dd>
          </dl>
          <button
            type="button"
            class="mt-6 w-full rounded-md border border-transparent bg-teal-600 py-2 px-4 text-sm font-medium text-white shadow-sm hover:bg-teal-700 focus:outline-none focus:ring-2 focus:ring-teal-500 focus:ring-offset-2"
          >
            Send message
          </button>
        </div>
      </div>
    </ComboboxOptions>

    <div v-if="query !== '' && filteredPeople.length === 0" class="py-14 px-6 text-center text-sm sm:px-14">
      <UsersIcon class="mx-auto h-6 w-6 text-gray-400" aria-hidden="true" />
      <p class="mt-4 font-semibold text-gray-900">No people found</p>
      <p class="mt-2 text-gray-500">We couldnâ€™t find anything with that term. Please try again.</p>
    </div>
  </Combobox>
</template>
