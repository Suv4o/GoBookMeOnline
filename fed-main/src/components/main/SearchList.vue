<script lang="ts">
export default {
  name: 'SearchList',
}
</script>

<script setup lang="ts">
import BookSearchList from '../global/BookSearchList/BookSearchList.vue'
import BookPagination from '../global/BookPagination/BookPagination.vue'
</script>

<template>
  <book-search-list title="Locations" type="location" />
  <book-search-list title="People" type="person" />
  <div class="mt-4">
    <book-pagination />
  </div>
</template>
