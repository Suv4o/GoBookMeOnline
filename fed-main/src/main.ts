import { createApp } from 'vue'
import App from './App.vue'
import router from './router'
import { createPinia } from 'pinia'
import './assets/tailwind/index.css'
import { auth } from './config/firebase.config'
import { onAuthStateChanged } from 'firebase/auth'
import { useAuthStore } from './store/auth'
export const pinia = createPinia()
const appConfig = { isAppMounted: false }

const app = createApp(App)
app.provide('$auth', auth)
app.use(pinia)
app.use(router)

onAuthStateChanged(auth, user => {
  if (!user) {
    const useStoreAuth = useAuthStore(pinia)
    useStoreAuth.accessToken = ''
    useStoreAuth.accessTokenExpirationTime = 0
    useStoreAuth.user = null
  }

  if (!appConfig.isAppMounted) {
    appConfig.isAppMounted = true
    router.isReady().then(() => {
      app.mount('#app')
    })
  }
})
