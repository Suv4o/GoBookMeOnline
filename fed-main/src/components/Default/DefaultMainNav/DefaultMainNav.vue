<script lang="ts">
export default {
  name: 'DefaultMainNav',
}
</script>

<script setup lang="ts">
import { Auth, signOut } from 'firebase/auth'
import { Popover, PopoverButton, PopoverPanel, Menu, MenuButton, MenuItem, MenuItems } from '@headlessui/vue'
import { Bars3Icon, XMarkIcon } from '@heroicons/vue/24/outline'
import { useAuthStore } from '../../../store/auth'
import useState from './useState'
import { inject } from 'vue'
import { Assertions } from '../../../types/guards'
import { useNotification } from '../../../utils/composables/notification'
import { NotificationTypes } from '../../../store/notification'
import { parseFirebaseError } from '../../../utils/helpers'
import SignupDropDownButton from './SignupDropDownButton.vue'
import { Roles } from '../../../types/enums'

const $auth = inject('$auth') as Auth
const { isSignInButtonShown, isSignUpButtonShown, setRoleType } = useState()

const signupOptions = [
  {
    name: 'Sign up for free',
    description: 'Find and organise your next booking online.',
    href: 'signup',
    roleType: Roles.USER_DEFAULT,
  },
  {
    name: 'Become a provider',
    description: 'Host your business online and get more bookings.',
    href: 'pricing-provider',
    roleType: Roles.PROVIDER_DEFAULT,
  },
]

async function signUserOut() {
  try {
    await signOut($auth)
  } catch (error) {
    Assertions.isError(error)
    const readableError = parseFirebaseError(error.message)
    if (readableError) {
      useNotification({ type: NotificationTypes.Error, title: 'Error', message: readableError })
    } else {
      useNotification({ type: NotificationTypes.Error, title: error.name, message: error.message })
    }
  }
}
</script>

<template>
  <Popover>
    <div class="max-w-7xl mx-auto px-4 sm:px-6">
      <nav class="relative flex items-center justify-between sm:h-10 md:justify-center" aria-label="Global">
        <div class="flex items-center flex-1 md:absolute md:inset-y-0 md:left-0">
          <div class="flex items-center justify-between w-full md:w-auto">
            <router-link :to="{ name: 'home' }" class="text-gray-500 hover:text-gray-600">
              <span class="sr-only">Go Book Me Online</span>
              <div class="flex items-center">
                <img class="h-8 w-auto sm:h-10" src="../../../assets/images/logo.svg" alt="" />
                <p class="sm:ml-4 ml-2 sm:text-lg font-semibold">GoBookMe.Today</p>
              </div>
            </router-link>
            <div class="-mr-2 flex items-center md:hidden">
              <PopoverButton
                data-testid="Open menu"
                class="bg-gray-50 rounded-md p-2 inline-flex items-center justify-center text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-teal-500"
              >
                <span class="sr-only">Open main menu</span>
                <Bars3Icon class="h-6 w-6" aria-hidden="true" />
              </PopoverButton>
            </div>
          </div>
        </div>
        <div
          v-if="useAuthStore().isUserReady"
          class="hidden md:absolute md:flex md:items-center md:justify-end md:inset-y-0 md:right-0"
        >
          <Menu
            v-if="isSignInButtonShown && useAuthStore().user && useAuthStore().userInitials"
            as="div"
            class="ml-3 relative"
          >
            <div>
              <MenuButton
                class="bg-white rounded-full flex text-sm focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-teal-500"
              >
                <span class="sr-only">Open user menu</span>
                <span class="inline-flex items-center justify-center h-12 w-12 rounded-full bg-teal-600">
                  <span data-testid="User Initials Desktop" class="text-lg font-medium leading-none text-white">{{
                    useAuthStore().userInitials
                  }}</span>
                </span>
              </MenuButton>
            </div>
            <transition
              enter-active-class="transition ease-out duration-200"
              enter-from-class="transform opacity-0 scale-95"
              enter-to-class="transform opacity-100 scale-100"
              leave-active-class="transition ease-in duration-75"
              leave-from-class="transform opacity-100 scale-100"
              leave-to-class="transform opacity-0 scale-95"
            >
              <MenuItems
                class="origin-top-right absolute right-0 mt-2 w-48 rounded-md shadow-lg py-1 bg-white ring-1 ring-black ring-opacity-5 focus:outline-none"
              >
                <MenuItem v-slot="{ active }">
                  <a href="#" :class="[active ? 'bg-gray-100' : '', 'block px-4 py-2 text-sm text-gray-700']"
                    >Your Profile</a
                  >
                </MenuItem>
                <MenuItem v-slot="{ active }">
                  <a href="#" :class="[active ? 'bg-gray-100' : '', 'block px-4 py-2 text-sm text-gray-700']"
                    >Settings</a
                  >
                </MenuItem>
                <MenuItem v-slot="{ active }">
                  <a
                    data-testid="Sign out"
                    href="javascript:;"
                    :class="[active ? 'bg-gray-100' : '', 'block px-4 py-2 text-sm text-gray-700']"
                    @click="signUserOut"
                    >Sign Out</a
                  >
                </MenuItem>
              </MenuItems>
            </transition>
          </Menu>
          <span v-else class="inline-flex rounded-md">
            <router-link
              v-if="isSignInButtonShown"
              data-testid="Sign in"
              :to="{ name: 'signin' }"
              class="inline-flex items-center px-4 py-2 border border-transparent text-base font-medium rounded-md text-teal-600 bg-white hover:bg-gray-50 shadow"
            >
              Sign in
            </router-link>
            <SignupDropDownButton v-if="isSignUpButtonShown" :signup-options="signupOptions" />
          </span>
        </div>
      </nav>
    </div>

    <transition
      enter-active-class="duration-150 ease-out"
      enter-from-class="opacity-0 scale-95"
      enter-to-class="opacity-100 scale-100"
      leave-active-class="duration-100 ease-in"
      leave-from-class="opacity-100 scale-100"
      leave-to-class="opacity-0 scale-95"
    >
      <PopoverPanel focus class="absolute z-10 top-0 inset-x-0 p-2 transition transform origin-top-right md:hidden">
        <div class="rounded-lg shadow-md bg-white ring-1 ring-black ring-opacity-5 overflow-hidden">
          <div class="px-5 pt-4 mb-4 flex items-center justify-between">
            <div class="flex items-center">
              <img class="h-8 w-auto" src="../../../assets/images/logo.svg" alt="" />
              <p class="ml-2 font-semibold">GoBookMe.Today</p>
            </div>
            <div class="-mr-2">
              <PopoverButton
                data-testid="Close menu"
                class="bg-white rounded-md p-2 inline-flex items-center justify-center text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-teal-500"
              >
                <span class="sr-only">Close menu</span>
                <XMarkIcon class="h-6 w-6" aria-hidden="true" />
              </PopoverButton>
            </div>
          </div>
          <div v-if="useAuthStore().isUserReady" class="px-5">
            <div v-if="isSignInButtonShown && useAuthStore().user && useAuthStore().userInitials" class="pb-4">
              <span class="inline-flex items-center justify-center h-10 w-10 rounded-full bg-teal-600 ml-3">
                <span data-testid="User Initials Mobile" class="text-lg font-medium leading-none text-white">{{
                  useAuthStore().userInitials
                }}</span>
              </span>
              <div class="space-y-1 mt-4" aria-label="Sidebar">
                <a
                  href="#"
                  class="text-gray-600 hover:bg-gray-50 hover:text-gray-900 flex items-center px-3 py-2 text-md font-medium rounded-md"
                >
                  <span class="truncate">Your Profile</span>
                </a>
                <a
                  href="#"
                  class="text-gray-600 hover:bg-gray-50 hover:text-gray-900 flex items-center px-3 py-2 text-md font-medium rounded-md"
                >
                  <span class="truncate">Settings</span>
                </a>
                <a
                  href="javascript:;"
                  class="text-gray-600 hover:bg-gray-50 hover:text-gray-900 flex items-center px-3 py-2 text-md font-medium rounded-md"
                  @click="signUserOut"
                >
                  <span class="truncate">Sign Out</span>
                </a>
              </div>
            </div>
            <template v-else>
              <router-link
                :to="{ name: 'signup' }"
                class="block w-full px-5 py-3 text-center font-medium text-white bg-teal-600 hover:bg-teal-700"
                @click="setRoleType(Roles.USER_DEFAULT)"
              >
                Sign up for free
              </router-link>
              <router-link
                :to="{ name: 'pricing-provider' }"
                class="block w-full px-5 py-3 text-center font-medium text-teal-600 bg-white hover:bg-gray-50 shadow my-2"
              >
                Become a provider
              </router-link>
              <p class="my-4 text-center text-base font-medium text-gray-500">
                Existing customer?
                <router-link data-testid="Sign in" :to="{ name: 'signin' }" class="text-teal-600 hover:text-teal-500">
                  Sign in
                </router-link>
              </p>
            </template>
          </div>
        </div>
      </PopoverPanel>
    </transition>
  </Popover>
</template>
