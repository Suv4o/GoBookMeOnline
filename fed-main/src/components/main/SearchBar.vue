<script lang="ts">
export default {
  name: 'SearchBar',
}
</script>

<script setup lang="ts">
import SearchInput from './SearchInput.vue'
import LocationInput from './LocationInput.vue'
</script>

<template>
  <div class="flex flex-row mx-auto max-w-3xl">
    <div class="basis-2/3 mr-4">
      <search-input />
    </div>
    <div class="basis-1/3"><location-input /></div>
  </div>
</template>
