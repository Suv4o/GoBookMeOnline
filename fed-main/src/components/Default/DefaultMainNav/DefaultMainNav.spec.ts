import { render, fireEvent, cleanup } from '@testing-library/vue'
import { describe, it, expect, vi, beforeAll } from 'vitest'
import { createRouter, createWebHistory, Router } from 'vue-router'
import DefaultMainNav from './DefaultMainNav.vue'
import { Auth, getAuth } from 'firebase/auth'
import * as dotenv from 'dotenv'
import { initializeFirebaseApp } from '../../../config/firebase.config'
import { pinia } from '../../../main'
import { useAuthStore } from '../../../store/auth'
dotenv.config({ path: `./env/env.${process.env.NODE_ENV}` })

let router: Router
let auth: Auth

beforeAll(() => {
  router = createRouter({
    history: createWebHistory(),
    routes: [
      {
        path: '/',
        name: 'home',
        components: {
          default: () => import('../../views/HomePage.vue'),
          DefaultMainNav: () => import('../../components/Default/DefaultMainNav/DefaultMainNav.vue'),
          DefaultNewsletter: () => import('../../components/Default/DefaultNewsletter/DefaultNewsletter.vue'),
          DefaultFooter: () => import('../../components/Default/DefaultFooter/DefaultFooter.vue'),
        },
      },
      {
        path: '/signin',
        name: 'signin',
        components: {
          default: () => import('../../views/SigninPage.vue'),
          DefaultMainNav: () => import('../../components/Default/DefaultMainNav/DefaultMainNav.vue'),
          DefaultNewsletter: () => import('../../components/Default/DefaultNewsletter/DefaultNewsletter.vue'),
          DefaultFooter: () => import('../../components/Default/DefaultFooter/DefaultFooter.vue'),
        },
      },
      {
        path: '/signup',
        name: 'signup',
        components: {
          default: () => import('../../views/SignupPage.vue'),
          DefaultMainNav: () => import('../../components/Default/DefaultMainNav/DefaultMainNav.vue'),
          DefaultNewsletter: () => import('../../components/Default/DefaultNewsletter/DefaultNewsletter.vue'),
          DefaultFooter: () => import('../../components/Default/DefaultFooter/DefaultFooter.vue'),
        },
      },
    ],
  })
  auth = getAuth(initializeFirebaseApp(process.env as ImportMetaEnv))
})

describe('DefaultMainNav', async () => {
  it('render component correctly', async () => {
    const wrapper = render(DefaultMainNav, {
      global: {
        plugins: [pinia, router],
        provide: {
          $auth: auth,
        },
      },
    })

    // Test if Desktop Signin and Signup buttons are rendered
    const buttonSignOutDesktop = wrapper.getAllByTestId('Sign up')
    const buttonSignInDesktop = wrapper.getAllByTestId('Sign in')

    expect(buttonSignOutDesktop.length).toBe(1)
    expect(buttonSignInDesktop.length).toBe(1)

    // Open Mobile Menu and test if both Desktop and Mobile Signin and Signup buttons are rendered
    const buttonHamburgerMenu = wrapper.getByTestId('Open menu')
    await fireEvent.click(buttonHamburgerMenu)

    const buttonSignOutDesktopMobile = wrapper.getAllByTestId('Sign up')
    const buttonSignInDesktopMobile = wrapper.getAllByTestId('Sign in')

    expect(buttonSignOutDesktopMobile.length).toBe(2)
    expect(buttonSignInDesktopMobile.length).toBe(2)

    cleanup()
  })

  it('render component with sign in user correctly', async () => {
    const useStoreAuth = useAuthStore(pinia)

    useStoreAuth.user = {
      displayName: 'Foo Bar',
      email: 'test@test.com',
      emailVerified: true,
      firstName: 'Foo',
      lastName: 'Bar',
      phoneNumber: null,
      photoURL: null,
      role: 'USER_DEFAULT',
      uid: 'ZU4jn14tbWfspKuWorshlqQ1s2E3',
    }

    useStoreAuth.isUserReady = true

    const wrapper = render(DefaultMainNav, {
      global: {
        plugins: [pinia, router],
        provide: {
          $auth: auth,
        },
      },
    })

    // Render User Initials currently on Desktop
    const userInitialsDesktop = wrapper.getByTestId('User Initials Desktop')
    expect(userInitialsDesktop.textContent).toBe(useStoreAuth.userInitials)

    // Open Mobile Menu and render User Initials currently on Mobile
    const buttonHamburgerMenu = wrapper.getByTestId('Open menu')
    await fireEvent.click(buttonHamburgerMenu)

    const userInitialsMobile = wrapper.getByTestId('User Initials Mobile')
    expect(userInitialsMobile.textContent).toBe(useStoreAuth.userInitials)

    cleanup()
  })

  it('snap shot matches', async () => {
    const wrapper = render(DefaultMainNav, {
      global: {
        plugins: [pinia, router],
        provide: {
          $auth: auth,
        },
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
