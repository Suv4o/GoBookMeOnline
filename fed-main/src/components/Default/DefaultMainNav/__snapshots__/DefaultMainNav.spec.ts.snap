// Vitest Snapshot v1

exports[`DefaultMainNav > snap shot matches 1`] = `
{
  "baseElement": <body>
    <div>
      <div>
        <div
          class="max-w-7xl mx-auto px-4 sm:px-6"
        >
          <nav
            aria-label="Global"
            class="relative flex items-center justify-between sm:h-10 md:justify-center"
          >
            <div
              class="flex items-center flex-1 md:absolute md:inset-y-0 md:left-0"
            >
              <div
                class="flex items-center justify-between w-full md:w-auto"
              >
                <a
                  class="text-gray-500 hover:text-gray-600"
                  href="/"
                >
                  <span
                    class="sr-only"
                  >
                    Go Book Me Online
                  </span>
                  <div
                    class="flex items-center"
                  >
                    <img
                      alt=""
                      class="h-8 w-auto sm:h-10"
                      src="/src/assets/images/logo.svg"
                    />
                    <p
                      class="sm:ml-4 ml-2 sm:text-lg font-semibold"
                    >
                      GoBookMe.Today
                    </p>
                  </div>
                </a>
                <div
                  class="-mr-2 flex items-center md:hidden"
                >
                  
                  <button
                    aria-expanded="false"
                    class="bg-gray-50 rounded-md p-2 inline-flex items-center justify-center text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-teal-500"
                    data-testid="Open menu"
                    id="headlessui-popover-button-20"
                    type="button"
                  >
                    <span
                      class="sr-only"
                    >
                      Open main menu
                    </span>
                    <svg
                      aria-hidden="true"
                      class="h-6 w-6"
                      fill="none"
                      stroke="currentColor"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M4 6h16M4 12h16M4 18h16"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                    </svg>
                  </button>
                  <!---->
                  
                </div>
              </div>
            </div>
            <div
              class="hidden md:absolute md:flex md:items-center md:justify-end md:inset-y-0 md:right-0"
            >
              <div
                class="ml-3 relative"
              >
                <div>
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="bg-white rounded-full flex text-sm focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-teal-500"
                    id="headlessui-menu-button-23"
                    type="button"
                  >
                    <span
                      class="sr-only"
                    >
                      Open user menu
                    </span>
                    <span
                      class="inline-flex items-center justify-center h-12 w-12 rounded-full bg-teal-600"
                    >
                      <span
                        class="text-lg font-medium leading-none text-white"
                        data-testid="User Initials Desktop"
                      >
                        FB
                      </span>
                    </span>
                  </button>
                </div>
                <transition-stub>
                  <!---->
                </transition-stub>
              </div>
            </div>
          </nav>
        </div>
        <transition-stub>
          <!---->
        </transition-stub>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="max-w-7xl mx-auto px-4 sm:px-6"
      >
        <nav
          aria-label="Global"
          class="relative flex items-center justify-between sm:h-10 md:justify-center"
        >
          <div
            class="flex items-center flex-1 md:absolute md:inset-y-0 md:left-0"
          >
            <div
              class="flex items-center justify-between w-full md:w-auto"
            >
              <a
                class="text-gray-500 hover:text-gray-600"
                href="/"
              >
                <span
                  class="sr-only"
                >
                  Go Book Me Online
                </span>
                <div
                  class="flex items-center"
                >
                  <img
                    alt=""
                    class="h-8 w-auto sm:h-10"
                    src="/src/assets/images/logo.svg"
                  />
                  <p
                    class="sm:ml-4 ml-2 sm:text-lg font-semibold"
                  >
                    GoBookMe.Today
                  </p>
                </div>
              </a>
              <div
                class="-mr-2 flex items-center md:hidden"
              >
                
                <button
                  aria-expanded="false"
                  class="bg-gray-50 rounded-md p-2 inline-flex items-center justify-center text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-teal-500"
                  data-testid="Open menu"
                  id="headlessui-popover-button-20"
                  type="button"
                >
                  <span
                    class="sr-only"
                  >
                    Open main menu
                  </span>
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4 6h16M4 12h16M4 18h16"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </button>
                <!---->
                
              </div>
            </div>
          </div>
          <div
            class="hidden md:absolute md:flex md:items-center md:justify-end md:inset-y-0 md:right-0"
          >
            <div
              class="ml-3 relative"
            >
              <div>
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="bg-white rounded-full flex text-sm focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-teal-500"
                  id="headlessui-menu-button-23"
                  type="button"
                >
                  <span
                    class="sr-only"
                  >
                    Open user menu
                  </span>
                  <span
                    class="inline-flex items-center justify-center h-12 w-12 rounded-full bg-teal-600"
                  >
                    <span
                      class="text-lg font-medium leading-none text-white"
                      data-testid="User Initials Desktop"
                    >
                      FB
                    </span>
                  </span>
                </button>
              </div>
              <transition-stub>
                <!---->
              </transition-stub>
            </div>
          </div>
        </nav>
      </div>
      <transition-stub>
        <!---->
      </transition-stub>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
